src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs

```diff
@@ -35,40 +35,7 @@ public static ValueTask<ImmutableArray<DiagnosticData>> GetPushDiagnosticsAsync(
             DiagnosticMode diagnosticMode,
             CancellationToken cancellationToken)
         {
-            return GetDiagnosticsAsync(service, document.Project.Solution.Workspace, document.Project, document, includeSuppressedDiagnostics, forPullDiagnostics: true, diagnosticMode, cancellationToken);
-        }
-
-        public static async ValueTask<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
-            this IDiagnosticService service,
-            Workspace workspace,
-            Project? project,
-            Document? document,
-            bool includeSuppressedDiagnostics,
-            bool forPullDiagnostics,
-            DiagnosticMode diagnosticMode,
-            CancellationToken cancellationToken)
-        {
-            Contract.ThrowIfTrue(document != null && document.Project != project);
-            Contract.ThrowIfTrue(project != null && project.Solution.Workspace != workspace);
-
-            using var _ = ArrayBuilder<DiagnosticData>.GetInstance(out var result);
-
-            var buckets = forPullDiagnostics
-                ? service.GetPullDiagnosticBuckets(workspace, project?.Id, document?.Id, diagnosticMode, cancellationToken)
-                : service.GetPushDiagnosticBuckets(workspace, project?.Id, document?.Id, diagnosticMode, cancellationToken);
```

Comment:
do we still need to expose teh buckets now?

Reply:
There's some other code that is still asking for buckets and using the bucket extension methods so that's not obviously deletable. I might also be needing it for some other stuff, so that'll be left around for now.

Reply:
![image](https://user-images.githubusercontent.com/9797472/163275211-a3d52843-974d-4a86-8830-03c808e9c62d.png)

Reply:
Oh I haven't seen that bucket meme in a long, long time...